packages/contracts/contracts/CyberMarketplace.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.9;
+
+import '@openzeppelin/contracts/utils/Counters.sol';
+import '@openzeppelin/contracts/utils/Address.sol';
+//import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
+import '@openzeppelin/contracts/utils/introspection/ERC165Checker.sol';
+import '@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol';
+
+bytes4 constant NFT_SIG = 0xd9b67a26;
+
+contract CyberMarketplace is ERC1155Holder {
+  using Counters for Counters.Counter;
+
+  struct Listing {
+    uint256 id;
+    address nftContract;
+    address payable seller;
+    uint256 tokenId;
+    uint256 price;
+    uint256 endTime;
+  }
+
+  event Sell(
+    uint256 id,
+    address nftContract,
+    uint256 tokenId,
+    address seller,
+    uint256 price,
+    uint256 endTime
+  );
+
+  event Buy(
+    uint256 id,
+    address nftContract,
+    uint256 tokenId,
+    address seller,
+    address buyer,
+    uint256 price
+  );
+
+  Counters.Counter private listingsId;
+
+  mapping(uint256 => Listing) private listings;
+
+  function sell(
+    address nftContract,
```

Comment:
make an init function to whitelist only specific contracts: we want to allow only onCyber contract (under the exclusive control of the contract owner)